website/static/js/accountSettings.js

```diff
@@ -89,13 +90,18 @@ var UserProfileClient = oop.defclass({
 
         return ret.promise();
     },
-    update: function (profile) {
+    update: function (profile, email) {
+        var self = this;
```

Comment:
There's no need to define `self` here, as `this` is bound in context at the end of the function definition.

If you keep `self`, let's from `.bind(this)` below.